app/src/main/java/com/babylon/wallet/android/presentation/transfer/prepare/PrepareManifestDelegate.kt

```diff
@@ -132,19 +132,19 @@ class PrepareManifestDelegate(
         }
     }
 
-    private suspend fun BabylonManifestBuilder.deposit(targetAccount: TargetAccount, bucket: ManifestBuilderBucket) =
-        apply {
-            val isUserAccount = targetAccount.isUserAccount
-            // TODO Temporary revert of checking if the receiving account is a ledger account
-            val isSoftwareAccount = true // !targetAccount.isLedgerAccount
-
-            val isAccountAbleToSign = targetAccount.factorSourceId?.let {
-                it.kind == FactorSourceKind.LEDGER_HQ_HARDWARE_WALLET ||
-                    (it.kind == FactorSourceKind.DEVICE && mnemonicRepository.mnemonicExist(it))
-            } ?: false
-
-            // we want to use try_deposit_or_abort for account that we are not controlling and are not able to sign tx
-            if (isUserAccount && isSoftwareAccount && isAccountAbleToSign) {
+    private suspend fun BabylonManifestBuilder.deposit(
+        targetAccount: TargetAccount,
+        bucket: ManifestBuilderBucket
+    ) = apply {
+        val isAccountAbleToSign = targetAccount.factorSourceId?.let {
+            it.kind == FactorSourceKind.LEDGER_HQ_HARDWARE_WALLET ||
+                (it.kind == FactorSourceKind.DEVICE && mnemonicRepository.mnemonicExist(it))
+        } ?: false
```

Comment:
I acutally I would like SOMEHOW to bring this information (if recovery is needed = mnemonic doesnt exist) with the accounts in the presentation layer. 
Maybe refactor the getprofileusecase in a way to return the accounts AND their `isRecoveryNeeded`, which is used also in other places? 
if you think about it `isRecoveryNeeded` is part of an account?!

Anyway I didn't find any good solution now. Maybe there is not at all. But keep it in mind

Reply:
> isRecoveryNeeded

I see what you mean yes 👍 Or similar easy win (not necessarily for this PR) might be just  `isRecoveryNeeded` backing field under Account namespace ?